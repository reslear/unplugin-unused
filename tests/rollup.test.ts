import path from 'node:path'
import {
  rollupBuild,
  RollupToStringPlugin,
  testFixtures,
} from '@sxzz/test-utils'
import { describe } from 'vitest'
import UnpluginUnused from '../src/rollup'

describe('rollup', async () => {
  const { dirname } = import.meta
  await testFixtures(
    '*.{js,tsx,vue}',
    async (args, id) => {
      const { snapshot } = await rollupBuild(id, [
        UnpluginUnused({
          level: 'error',
        }),
        RollupToStringPlugin(),
      ])
      return snapshot
    },
    {
      cwd: path.resolve(dirname, 'fixtures'),
      promise: true,
    },
  )
})
