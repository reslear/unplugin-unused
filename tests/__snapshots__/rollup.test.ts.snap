// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rollup > basic.js 1`] = `
"// basic.js
var basic = \`import '@rollup/pluginutils'
import 'js-tokens'
import 'pkg-types'
import 'unplugin'
\`;

export { basic as default };
"
`;

exports[`rollup > error.js 1`] = `[Error: Unused dependencies found: js-tokens]`;

exports[`rollup > error-tsx.tsx 1`] = `[Error: Unused dependencies found: unplugin]`;

exports[`rollup > error-vue.vue 1`] = `[Error: Unused dependencies found: @rollup/pluginutils, js-tokens, pkg-types]`;
